source/common/config/new_delta_subscription_state.cc

```diff
@@ -172,7 +172,7 @@ bool NewDeltaSubscriptionState::isHeartbeatResponse(
   }
 
   if (const auto itr = ambiguous_resource_state_.find(resource.name());
-      itr != wildcard_resource_state_.end()) {
+      itr != ambiguous_resource_state_.end()) {
```

Comment:
Makes sense, can you add a test that would have caught this?

Reply:
I don't think that technically this is possible. In all my tests `wildcard_resource_state_.end()` and `ambiguous_resource_state_.end()` are always equal.
I'm assuming that this is because a the iterator points to some "null" location, that `end()` just returns, but will need to verify this through the abseil code if requested.

Reply:
Sure, just wanted to check, thanks.